compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -913,7 +913,23 @@ trait Applications extends Compatibility {
 
       fun1.tpe match {
         case err: ErrorType => cpy.Apply(tree)(fun1, proto.typedArgs()).withType(err)
-        case TryDynamicCallType => typedDynamicApply(tree, pt)
+        case TryDynamicCallType =>
+          val isInsertedApply = fun1 match {
+            case Select(_, nme.apply) => fun1.span.isSynthetic
+            case TypeApply(sel @ Select(_, nme.apply), _) => sel.span.isSynthetic
+            /* TODO Get rid of this case. It is still syntax-based, therefore unreliable.
+             * It is necessary for things like `someDynamic[T](...)`, because in that case,
+             * somehow typedFunPart returns a tree that was typed as `TryDynamicCallType`,
+             * so clearly with the view that an apply insertion was necessary, but doesn't
+             * actually insert the apply!
+             * This is probably something wrong in apply insertion, but I (@sjrd) am out of
+             * my depth there.
+             * In the meantime, this makes tests pass.
+             */
+            case TypeApply(fun, _) => !fun.isInstanceOf[Select]
```

Comment:
Shouldn't we also check that the name of `fun` has a synthetic span?

Reply:
Oh, no

Reply:
What kind of trees do we get here?

Reply:
Anything. For the test that was failing, it was `TypeApply(Ident("qual"), List(oneTArg))`.

Reply:
Where I was expecting `typedFunPart` to have returned `TypeApply(Select(Ident("qual"), nme.apply), List(oneTarg))` instead, where the `Select` node would have had a synthetic span.

Reply:
I'm wondering if we can get a `TypeApply(Block(..., Select(..., nme.apply)), ...)`. I guess those would never be inserted applies.

Reply:
No, in that case, for an inserted apply, the `Select` should be around the `Block`, not inside it.